Makefile

```diff
@@ -12,7 +12,7 @@ RELEASE_BUILD ?= false
 CLOUD_PROVIDER ?=
 GOOPTIONS   ?= GOOS=linux GOARCH=$(ARCH) CGO_ENABLED=0
 GOFLAGS     ?=
-BINARIES    := cloud-api-adaptor agent-protocol-forwarder
+BINARIES    := cloud-api-adaptor agent-protocol-forwarder process-user-data
```

Comment:
Shouldn't it be added for Azure only?

Reply:
@snir911 this is a generic command addition which includes two sub commands. One  to process user-data to update the agent config and another to fetch the metadata service from metadata service.

The default mechanism to fetch the user-data is still via config-drive.  And if metadata fetching is needed then the image needs to be built with DISABLE_CLOUD_CONFIG=true. 
For updating the agent-config using the user-data from the file is common for all providers.